src/app/static/src/app/store/adr/actions.ts

```diff
@@ -61,41 +68,54 @@ export const actions: ActionTree<ADRState, RootState> & ADRActions = {
             .delete("/adr/key/")
     },
 
-    async getDatasets(context) {
-        context.commit({type: ADRMutation.SetFetchingDatasets, payload: true});
-        context.commit({type: ADRMutation.SetADRError, payload: null});
+    async getDatasets(context, datasetType: AdrDatasetType) {
+        context.commit({type: ADRMutation.SetFetchingDatasets, payload: {datasetType, data: true}});
+        context.commit({type: ADRMutation.SetADRError, payload: {datasetType, data: null}});
         await api<ADRMutation, ADRMutation>(context)
-            .withError(ADRMutation.SetADRError)
-            .withSuccess(ADRMutation.SetDatasets)
+            .withErrorCallback((failure: Response) => {
+                const error = APIService.getFirstErrorFromFailure(failure);
+                context.commit({type: ADRMutation.SetADRError, payload: {datasetType, data: error}});
+            })
```

Comment:
i dont think this works, i tried importing the only release for ethiopia but from my console logs it was using the original `this._onError` function

Reply:
rip :laughing: